daemon: move map-related code to map.go

ianvernon opened the pull request:
Navigating daemon.go is confusing; there is a lot of code in various places that
could be separated out into smaller files that more logically indicate the
function of the code contained inside of them. Move map-related code to its own
file accordingly.

Signed-off by: Ian Vernon <ian@cilium.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5442)
<!-- Reviewable:end -->

joestringer reviewed (commented):
FWIW We are proposing to remove most of this code in the following issue: https://github.com/cilium/cilium/issues/5168#issuecomment-413474441

aanm reviewed (dismissed):
Waiting for addressing Thomas comment

Commit:
daemon: move map-related code to map.go

Navigating daemon.go is confusing; there is a lot of code in various places that
could be separated out into smaller files that more logically indicate the
function of the code contained inside of them. Move map-related code to its own
package within `pkg/datapath`.

Signed-off by: Ian Vernon <ian@cilium.io>

ianvernon commented on pkg/datapath/maps/map.go:
```diff
@@ -0,0 +1,106 @@
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "datapath-maps")
+)
+
+func CollectStaleMapGarbage() {
```
Resolved in next upload.

joestringer commented on pkg/datapath/maps/map.go:
```diff
@@ -0,0 +1,106 @@
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/endpointmanager"
```
These endpoint dependencies seem out-of-place.

ianvernon replied:
I noticed that too. The only values being used here are constants. These can be factored out into another package within `pkg/endpoint`.

joestringer reviewed (approved):
Seems fine to me, one minor comment of something that seems out of place but we can address such issues over time.
